src/topics/lda_model.cpp

```diff
@@ -16,30 +17,23 @@ lda_model::lda_model(const learn::dataset& docs, std::size_t num_topics)
     /* nothing */
 }
 
-void lda_model::save_doc_topic_distributions(const std::string& filename) const
+void lda_model::save_doc_topic_distributions(std::ostream& stream) const
 {
-    std::ofstream file{filename};
-    for (const auto& doc : docs_)
+    io::packed::write(stream, docs_.size());
+    io::packed::write(stream, num_topics_);
+
+    for (const auto& d : docs_)
     {
-        file << doc.id << "\t";
-        double sum = 0;
+        auto dist = topic_distrbution(doc_id{d.id});
         for (topic_id j{0}; j < num_topics_; ++j)
         {
-            double prob = compute_doc_topic_probability(doc.id, j);
-            if (prob > 0)
-                file << j << ":" << prob << "\t";
-            sum += prob;
+            io::packed::write(stream, dist.counts(j));
```

Comment:
Did you mean `dist.probability(j)`?

In fact, can't this just be `io::packed::write(stream, dist)` instead? `stats::multinomial<T>` implements the `io::packed` overloads.

This whole thing would then be something like:

```cpp
io::packed::write(stream, docs_.size());
for (const auto& d : docs_)
{
    io::packed::write(stream, topic_distribution(doc_id{d.id}));
}
```